File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
/**
 * 品牌控制层
 * @author Administrator
 *
 */

import entity.PageResult;
import entity.ResultMsg;
@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	/**
	 * 查询所有品牌信息
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	}
	
	/**
	 * 查询所有品牌信息-分页
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(Integer pageNum,Integer pageSize) {
		return brandService.findPage(pageNum, pageSize);
	}
	
	/**
	 * 添加品牌信息
	 * @param brand
	 */
	@RequestMapping("/add")
	public ResultMsg add(@RequestBody TbBrand brand) {
		try {
			brandService.add(brand);
			return new ResultMsg(true, "添加成功!");
		} catch (Exception e) {
			e.printStackTrace();
			return new ResultMsg(false, "添加失败!");
		}
	}
	
	/**
	 * 根据id查询品牌信息
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id) {
		return brandService.findOne(id);
	}
	
	/**
	 * 根据id修改信息
	 * @param brand
	 */
	@RequestMapping("/update")
	public ResultMsg update(@RequestBody TbBrand brand) {
		try {
			brandService.update(brand);
			return new ResultMsg(true, "修改成功!");
		} catch (Exception e) {
			e.printStackTrace();
			return new ResultMsg(false, "修改失败!");
		}
	}
	
	/**
	 * 根据id数组删除品牌信息
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public ResultMsg delete(Long[] ids) {
		try {
			brandService.delete(ids);
			return new ResultMsg(true, "删除成功!");
		} catch (Exception e) {
			e.printStackTrace();
			return new ResultMsg(false, "删除失败!");
		}
	}
	
	/**
	 * 模糊查询-分页
	 * @param brand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand,Integer pageNum,Integer pageSize) {
		return brandService.findPage(brand, pageNum, pageSize);
	}
	
	
	
	
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//定义控制器
    	app.controller("brandController",function($scope,$http,$controller,brandService){
    		//继承
    		$controller('baseController',{$scope:$scope});//继承 
    		
    		// 定义方法-查询所有-为分页
    		$scope.findAll=function(){
    			// 向后台发送请求
    			brandService.findAll().success(
    				// 接收后台发过来的数据
    				function(response){
    					$scope.list=response;
    				}
    			)
    		};
    		
    		//查询所有-分页
    		$scope.findPage=function(pageNum,pageSize){
    			//发送后台请求
    			brandService.findPage(pageNum,pageSize).success(
    				//接收请求		
    				function(response) {
    					//每页显示的集合
						$scope.list=response.rowsList;
    					//更新总记录数
						$scope.paginationConf.totalItems=response.total;
					}
    			)
    			
    		};
    
    		// 添加品牌信息
    		$scope.save=function(){
    			var　object = null;
    			if($scope.entity.id==null){
    				object=brandService.add($scope.entity);
    			}else{
    				object=brandService.update($scope.entity);
    			}
    			object.success(
	    				function(response){
							if(response.flag){
								$scope.reloadList();
								alert(response.message);
							}else{
								alert(response.message);
							}    					
	    				}		
	    			);    			
    		}
    		
    		//根据id查找品牌信息
    		$scope.findOne=function(id){
    			brandService.findOne(id).success(
    				function(response){
    					$scope.entity=response;
    				}		
    			);
    		}

    		//删除
    		$scope.delSelected=function(){
    			
    			//如果数组长度大于0
    			if($scope.ids.length>0){
    				if(confirm("您确定要删除所选信息么?")){
    					brandService.delSelected($scope.ids).success(
        	    				function(response){
        	    					if(response.flag){
        	    						$scope.reloadList();
        	    						alert(response.message);
        	    					}else{
        	    						alert(response.message);
        	    					}
        	    				}
        	    		)
        			}
    			}
    			
    		}
    		
    		//定义搜索对象
    		$scope.searchEntity={};
    		
    		//模糊查询
    		$scope.search=function(pageNum,pageSize){
    			brandService.search(pageNum,pageSize,$scope.searchEntity).success(
    				function(response){
    					$scope.list=response.rowsList;
    					$scope.paginationConf.totalItems=response.total;
    				}		
    			);
    		}
    })